internal/controller/nnf_node_controller.go

```diff
@@ -485,6 +629,232 @@ func updateDrives(node *nnfv1alpha9.NnfNode, log logr.Logger) error {
+			log.Error(err, "Failed to remove fence request file", "file", filePath)
+		} else {
+			log.Info("Removed processed fence request file", "file", filePath)
+		}
+	}
+}
+
+// watchFenceResponses watches the fence response directory and triggers NnfNode reconciliation
+// when response files are written (fencing) or deleted (unfencing). This allows the NnfNode to
+// update compute node fenced status.
+func (r *NnfNodeReconciler) watchFenceResponses(ctx context.Context, log logr.Logger) {
+	log = log.WithValues("component", "fenceResponseWatcher")
+	log.Info("Started watching fence response directory", "dir", fence.ResponseDir)
+
+	for {
+		select {
+		case <-ctx.Done():
+			log.Info("Stopping fence response watcher")
+			r.responseWatcher.Close()
+			return
+
+		case fsEvent, ok := <-r.responseWatcher.Events:
+			if !ok {
+				return
+			}
+
+			// Response write indicates fence operation complete, delete indicates unfencing
+			if fsEvent.Op&(fsnotify.Write|fsnotify.Remove) != 0 {
+				filename := filepath.Base(fsEvent.Name)
+				action := "written"
+				if fsEvent.Op&fsnotify.Remove != 0 {
+					action = "removed"
+				}
+				log.Info("Fence response file event, triggering NnfNode reconcile", "file", filename, "action", action)
+
+				// Trigger reconciliation of this NnfNode
+				r.Events <- event.GenericEvent{Object: &nnfv1alpha9.NnfNode{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      r.Name,
+						Namespace: r.Namespace,
+					},
+				}}
+			}
+
+		case err, ok := <-r.responseWatcher.Errors:
+			if !ok {
+				return
+			}
+			log.Error(err, "Fence response watcher error")
+		}
+	}
+}
+
+// triggerBlockStorageReconciles finds all GFS2 NnfNodeBlockStorage resources that grant access
+// to the given compute node and triggers their reconciliation
+func (r *NnfNodeReconciler) triggerBlockStorageReconciles(ctx context.Context, computeNode string, log logr.Logger) int {
+	// List all NnfNodeBlockStorage resources in this namespace
+	blockStorageList := &nnfv1alpha9.NnfNodeBlockStorageList{}
+	if err := r.List(ctx, blockStorageList, client.InNamespace(r.Namespace)); err != nil {
+		log.Error(err, "Failed to list NnfNodeBlockStorage resources")
+		return 0
+	}
+
+	log.Info("Listing NnfNodeBlockStorage resources", "count", len(blockStorageList.Items))
+
+	triggered := 0
+	for _, nnfNodeBlockStorage := range blockStorageList.Items {
+		// Only process GFS2 filesystems
+		allocationSet, hasLabel := nnfNodeBlockStorage.Labels["nnf.cray.hpe.com/allocationset"]
+		if !hasLabel || allocationSet != "gfs2" {
+			log.Info("Skipping non-GFS2 resource", "name", nnfNodeBlockStorage.Name, "allocationSet", allocationSet)
+			continue
+		}
+
+		// Check if this NnfNodeBlockStorage has allocations with access from the compute node
+		hasAccess := false
+		for _, allocation := range nnfNodeBlockStorage.Status.Allocations {
+			if allocation.Accesses != nil {
+				if _, found := allocation.Accesses[computeNode]; found {
+					hasAccess = true
```

Comment:
If you've already found the NnfNodeBlockStorage and seen that it has a StorageGroup for the fenced compute node, can you just do the DeleteStorageGroup here instead of triggering a Reconcile of the NnfNodeBlockStorage?